src/python/pants_release/generate_github_workflows.py

```diff
@@ -875,15 +875,19 @@ def build_wheels_job(
                             #   support. `curl` is a good lowest-common-denominator way to upload the assets.
                             "run": dedent(
                                 """\
-                                LOCAL_TAG=$(PEX_INTERPRETER=1 dist/src.python.pants/pants-pex.pex -c "import sys;major, minor = sys.version_info[:2];import os;uname = os.uname();print(f'cp{major}{minor}-{uname.sysname.lower()}_{uname.machine.lower()}')")
-                                mv dist/src.python.pants/pants-pex.pex dist/src.python.pants/pants.$LOCAL_TAG.pex
+                                PANTS_VER=$(PEX_INTERPRETER=1 dist/src.python.pants/pants-pex.pex -c "import pants.version;print(pants.version.VERSION)")
+                                PY_VER=$(PEX_INTERPRETER=1 dist/src.python.pants/pants-pex.pex -c "import sys;print(f'cp{sys.version_info[0]}{sys.version_info[1]}')")
+                                PLAT=$(PEX_INTERPRETER=1 dist/src.python.pants/pants-pex.pex -c "import os;print(f'{os.uname().sysname.lower()}_{os.uname().machine.lower()}')")
+                                PEX_FILENAME=pants.$PANTS_VER-$PY_VER-$PLAT.pex
+
+                                mv dist/src.python.pants/pants-pex.pex dist/src.python.pants/$PEX_FILENAME
 
                                 curl -L --fail \\
                                     -X POST \\
                                     -H "Authorization: Bearer ${{ github.token }}" \\
                                     -H "Content-Type: application/octet-stream" \\
-                                    ${{ needs.release_info.outputs.release-asset-upload-url }}?name=pants.$LOCAL_TAG.pex \\
-                                    --data-binary "@dist/src.python.pants/pants.$LOCAL_TAG.pex"
+                                    ${{ needs.release_info.outputs.release-asset-upload-url }}?name=$PEX_FILENAME \\
+                                    --data-binary "@dist/src.python.pants/$PEX_FILENAME"
```

Comment:
For supporting existing scie-pants installations, for 2.18, we could duplicate this to upload the old file. This makes things less convenient for ourselves for 2.18, but better for users (and we can remove for 2.19), that is, include an additional `curl` invocation with different `name`:


```
                                curl -L --fail \\
                                     -X POST \\
                                     -H "Authorization: Bearer ${{ github.token }}" \\
                                     -H "Content-Type: application/octet-stream" \\
                                     ${{ needs.release_info.outputs.release-asset-upload-url }}?name=pants.$PY_VER-$PLAT.pex \\
                                     --data-binary "@dist/src.python.pants/$PEX_FILENAME"
```

Reply:
What's the difference between 2.18 and 2.19 here? Users (outside of a select few) haven't seen the upgrade error, and as you said most of them will jump to latest, which _should_ include this change.

Reply:
Any users who've installed scie-pants 0.10 in the last few days will not be prompted to jump to latest when they start using 2.17.0.

Most importantly, I think it's more than just a select few: people newly adopting pants, people prompted by validating 2.17.0rc5, or even just seeing the scie-pants release announcements in #announce.

After this, when they upgrade to 2.18.0, the scie-pants 0.10 they've already got on their machine will fail hard about not finding the release asset, rather than providing a useful error message about upgrading.

Reply:
And just to be clear, you're suggesting we remedy that by:
- Having all of the Pants 2.18 release series upload both named PEXs
- Having Pants 2.18.x release series warn about needing to bump scie-pants to 0.10.N (whatever version starts referencing the new name)

Reply:
Yeah, exactly 👍

Reply:
I'm really torn, then.

On one hand, I agree that would be a nice path forward.

On the other hand, this journey is supposed to be about simplification. And uploading twice the amount of massive files, and polluting our asset lists doesn't give me the warm fuzzies.

Given that the users will get an error (instead of say, undefined behavior) I lean towards the ask-for-forgiveness route. Treat it like a "known bug", and tell people to upgrade.

Maybe I'm alone here though. Let's let others chime in? CC @stuhood @benjyw?

Reply:
(FWIW, I'm _strongly_ in the camp of us-as-pants-developers taking on annoyance to reduce annoyance for users, especially when it's temporary (just one release cycle), and just annoying to computers (not manual work). The impact on having slightly more release assets for 2.18 is annoying, but they're effectively an implementation detail. Causing confusion/frustration right on start-up seems like a particularly bad time to do so, too. Good diagnostics are awesome!)

Reply:
Ok. I'm becoming a believer.

We'll upload both for 2.18, and then 2.19 will just be versioned.

We'll bump the minimum version of scie-pants required in `main` and cherry pick to 2.17, but not merge it unless we're sure there's another RC, or until after the release is cut.